joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/datastructures/RubyScope.scala

```diff
@@ -371,4 +371,10 @@ class RubyScope(summary: RubyProgramSummary, projectRoot: Option[String])
       }
   }
 
+  def findFileNode(): Option[NewFile] = {
```

Comment:
Any reason that this is in the scope object instead of something like `protected val fileNode: Option[File] = None` in the `AstCreator`? My reasoning against this is that this adds another class and some stack complexity, which may not be necessary given it's always 1 file for a given `AstCreator`.

Reply:
Then you could do something like `fileNode.foreach(fn => diffGraph.addEdge(fn, namespaceBlock, EdgeTypes.AST))`